### PR TITLE
[TT-9635] Document values.yaml fields

### PR DESCRIPTION
**Pull request by buraksekili**
## Description

This PR adds documentation about fields in each values.yaml file

## Related Issue
https://tyktech.atlassian.net/browse/TT-9635

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tyk-stack/values.yaml [hunk 1]
```diff
@@ -818,19 +896,22 @@ tyk-dashboard:
     # - TYK_DB_SERVEROPTIONS_CERTIFICATE_CERTFILE
     # - TYK_DB_SERVEROPTIONS_CERTIFICATE_KEYFILE
     # variables in extraEnvs object array to define your SSL cert and key files.
```
Comment (caroltyk) [inline tyk-stack/values.yaml:898]: This part can be removed as not required 🙏

Comment (caroltyk) [reply tyk-stack/values.yaml:898]: also https://github.com/TykTechnologies/tyk-charts/blob/c0bc0e52e88667cdecc05569a5b213f2d0c04d31/tyk-stack/values.yaml#L57

File: components/tyk-bootstrap/values.yaml [hunk 2]
```diff
@@ -127,6 +136,9 @@ bootstrap:
   #     key: value
   #     post: install
   podAnnotations:
+    # preInstallPodAnnotations specifies annotations to be added Tyk Bootstrap pre-install hook's job pods.
```
Comment (komalsukhani) [inline components/tyk-bootstrap/values.yaml:139]:
```suggestion
    # preInstallPodAnnotations specifies annotations to be added to Tyk Bootstrap pre-install hook's job pods.
```

File: tyk-stack/values.yaml [hunk 3]
```diff
@@ -22,13 +22,16 @@ global:
     # - adminUserPassword - sets .global.adminUser.password
     useSecretName: ""
 
-    # First name of the admin user
+    # firstName corresponds to the first name of the admin user going to be created in bootstrap.
```
Comment (komalsukhani) [inline tyk-stack/values.yaml:25]:
```suggestion
    # firstName corresponds to the first name of the admin user that will be created during the bootstrapping process.
```

File: tyk-stack/values.yaml [hunk 4]
```diff
@@ -22,13 +22,16 @@ global:
...
+    # It is used to set TYK_K8SBOOTSTRAP_TYK_ADMIN_FIRSTNAME
     firstName: admin
 
-    # Last name of the admin user
+    # lastName corresponds to the last name of the admin user going to be created in bootstrap.
```
Comment (komalsukhani) [inline tyk-stack/values.yaml:29]:
```suggestion
    # lastName corresponds to the last name of the admin user that will be created during the bootstrapping process.
```

File: tyk-stack/values.yaml [hunk 5]
```diff
@@ -22,13 +22,16 @@ global:
...
+    # It is used to set TYK_K8SBOOTSTRAP_TYK_ADMIN_LASTNAME
     lastName: user
 
-    # Email address of the admin user
+    # email corresponds to the email of the admin user going to be created in bootstrap.
```
Comment (komalsukhani) [inline tyk-stack/values.yaml:33]:
```suggestion
    # email corresponds to the email of the admin user that will be created during the bootstrapping process.
```

File: tyk-stack/values.yaml [hunk 6]
```diff
@@ -696,8 +708,11 @@ tyk-bootstrap:
     #     key: value
     #     post: install
     podAnnotations:
+      # preInstallPodAnnotations specifies annotations to be added Tyk Bootstrap pre-install hook's job pods.
```
Comment (komalsukhani) [inline tyk-stack/values.yaml:711]:
```suggestion
      # preInstallPodAnnotations specifies annotations to be added to Tyk Bootstrap pre-install hook's job pods.
```

File: tyk-stack/values.yaml [hunk 7]
```diff
@@ -696,8 +708,11 @@ tyk-bootstrap:
...
       preInstallPodAnnotations: {}
+      # postInstallPodAnnotations specifies annotations to be added Tyk Bootstrap post-install hook's job pods.
```
Comment (komalsukhani) [inline tyk-stack/values.yaml:713]:
```suggestion
      # postInstallPodAnnotations specifies annotations to be added to Tyk Bootstrap post-install hook's job pods.
```

File: tyk-stack/values.yaml [hunk 8]
```diff
@@ -696,8 +708,11 @@ tyk-bootstrap:
...
       postInstallPodAnnotations: {}
+      # preDeletePodAnnotations specifies annotations to be added Tyk Bootstrap pre-delete hook's job pods.
```
Comment (komalsukhani) [inline tyk-stack/values.yaml:715]:
```suggestion
      # preDeletePodAnnotations specifies annotations to be added to Tyk Bootstrap pre-delete hook's job pods.
```

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: components/tyk-dashboard/values.yaml [hunk 9]
```diff
@@ -136,32 +146,83 @@ global:
...
   oauthRedirectUriSeparator: ";"
+  # TYK_DB_KEYREQUESTFIELDS
   keyRequestFields: "appName;appType"
+  # Dashboard session lifetime
+  # It is used to set TYK_DB_DASHBOARDSESSIONLIFETIME
   dashboardSessionLifetime: 43200
+  # When enabled, if dashboard already have user with given email found, it will be used for the login process
+  # It is used to set TYK_DB_SSOENABLEUSERLOOKUP
   ssoEnableUserLookup: true
+  # TYK_DB_NOTIFICATIONSLISTENPORT
```
[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-dashboard/values.yaml:213]: same as https://github.com/TykTechnologies/tyk-charts/pull/188#discussion_r1411764042

File: components/tyk-dashboard/values.yaml [hunk 10]
```diff
@@ -136,32 +146,83 @@ global:
...
+    # It is used to set TYK_DB_HOSTCONFIG_GATEWAYHOSTNAME
     overrideHostname: "tyk-gw.local"
+  # The path to the home directory of Tyk Dashboard, this must be set in order for Portal templates and other files to be loadable.
+  # It is used to set TYK_DB_HOMEDIR
   homeDir: "/opt/tyk-dashboard"
+  # If using the mongo-pump-selective pump, where data is written to org-id-specific collections in MongoDB,
+  # then enabling this option will switch querying for analytics over to the independent collection entries.
+  # It is used to set TYK_DB_USESHARDEDANALYTICS
   useShardedAnalytics: false
+  # It is used to set TYK_DB_ENABLEAGGREGATELOOKUPS
```
[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-dashboard/values.yaml:198]: added in c8690fb9f8c42c5489b7daf4e9f36276bd6901bf. can you please check again?

File: tyk-stack/values.yaml (hunk 1 above)
Comment (buraksekili) [reply tyk-stack/values.yaml]: thanks! removed in https://github.com/TykTechnologies/tyk-charts/pull/188/commits/35f7783f5898404799ecbbe700c4c54ccab01b74

File: components/tyk-dashboard/values.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-dashboard/values.yaml]: @komalsukhani updated in https://github.com/TykTechnologies/tyk-charts/pull/188/commits/f3f0b9c89361de88512eba075a74877132483808

Thanks for the help!

File: components/tyk-dashboard/values.yaml [hunk 11]
```diff
@@ -136,32 +146,83 @@ global:
...
   enableAggregateLookups: true
+  # TYK_DB_ENABLEANALYTICSCACHE
```
[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-dashboard/values.yaml]: addressed in https://github.com/TykTechnologies/tyk-charts/pull/188/commits/f3f0b9c89361de88512eba075a74877132483808

File: components/tyk-dashboard/values.yaml [hunk 12]
```diff
@@ -72,11 +76,17 @@ global:
...
+    # port corresponds to the port of postgres
     port: 5432
+    # user corresponds to the user of postgres
     user: postgres
+    # password corresponds to the password of the given postgres user in selected database
     password:
+    # database corresponds to the database to be used in postgres
     database: tyk_analytics
+    # sslmode corresponds to if postgres runs in sslmode (https)
     sslmode:
```
[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-dashboard/values.yaml:86]: https://tyktech.atlassian.net/browse/TT-10733

**Pull request merged**